### PR TITLE
App Engine running over Terraform

### PR DESCRIPTION
**Pull request by brunobowden**
Terraform:
- Terraform lint checks fixed (previously always skipped)
- developer-guide.md updated instructions

Server
- Removed unused code
- Deploy allowed for GCP project of any name
- Deploy consolidated to single script + server verification
- server-verify.sh terraform checks, works from any current directory

Tested
- In production on newly created project
- Cron jobs work
- Redirect for https://who-myhealth2-staging-bruno.appspot.com/app

## Checklist:

- [x] Followed the [Contributor Guidelines](https://github.com/WorldHealthOrganization/app/blob/master/docs/CONTRIBUTING.md) and verified that all contributions are properly licensed pursuant to the [LICENSE](https://github.com/WorldHealthOrganization/app/blob/master/LICENSE).

### CODE REVIEW

**Review by theswerd — APPROVED** [review]
LGTM

File: server/appengine/src/main/java/who/Environment.java [hunk 1]
```diff
@@ -7,10 +7,11 @@
  */
 public enum Environment {
   TEST("http://localhost:3000"),
-  DEVELOPMENT("http://localhost:3000"),
+  DEV_LOCAL("http://localhost:3000"),
```
[the remark this answers is not in the document]
Comment (brunobowden) [reply server/appengine/src/main/java/who/Environment.java:10]: To distinguish between local and remote servers. With "Dev" or "Development", it's ambiguous whether the server is running on localhost or in the cloud. By adding the suffix for "_local" or "_server", it's clearer which is which. Though maybe it should be `DEV_CLOUD`?

File: tools/verify-server.sh [hunk 2]
```diff
@@ -1,4 +1,10 @@
 #!/bin/sh
 set -euv
 cd ../server
+
+# Test
 ./gradlew test
+
+# Terraform
+cd ./terraform
+terraform validate .
```
[the remark this answers is not in the document]
Comment (brunobowden) [reply tools/verify-server.sh:10]: As a "verify" script, it shouldn't be doing any formatting but can check that the existing formatting is correct. This seems a little redundant but added:
```sh
terraform fmt -check -recursive .
```

**Comment by brunobowden** [comment]
@yukuairoy - I need your approval since this changes server code

**Review by yukuairoy — APPROVED** [review]

**Comment by brunobowden** [comment]
Thanks @yukuairoy. Changed to `terraform apply` and added in warning about not deleting resources as it may bork App Engine.

**Pull request merged**